### ISSUE TITLE
Mac Framework dependencies are not being discovered.

### ISSUE THREAD

**Issue by augustocdias** [microsoft/vscode-cpptools#1913]
I cannot make IntelliSense recognize the Mac OS frameworks... 
My application makes use of the ApplicationServices framework and IntelliSense cannot find `AE/AE.h` header for example. The only way I found to make it work was creating a symlink to the Headers directory from the framework. I'm using VS Code 1.22.2 and the C++ extension 0.16.1
Here's my properties:

        {
            "name": "Mac",
            "includePath": [
                "${workspaceFolder}",
                "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1",
                "/usr/local/include",
                "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include",
                "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include",
                "/usr/include",
                "${workspaceFolder}/import/include"
            ],
            "defines": [],
            "intelliSenseMode": "clang-x64",
            "browse": {
                "path": [
                    "${workspaceFolder}",
                    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1",
                    "/usr/local/include",
                    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.1.0/include",
                    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include",
                    "/usr/include"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            },
            "macFrameworkPath": [
                "/System/Library/Frameworks",
                "/Library/Frameworks"
            ],
            "compilerPath": "/usr/bin/clang++",
            "cStandard": "c11",
            "cppStandard": "c++17"
        }

**Comment by sean-mcmanus** [comment]
I'm not sure if this is a bug or not, but we can investigate later.

Do you know if this is a regression? 0.16.0 made changes related to the compilerPath that could have regressed this, so you could try the older 0.15.0 https://github.com/Microsoft/vscode-cpptools/releases/tag/v0.15.0 (but compilerPath/cStandard/cppStandard won't be recognized).

This is the first time I've seen "9.1.0" of clang/xcode being used instead of "9.0.0", so maybe there was a breaking change there.

**Comment by augustocdias** [comment]
@sean-mcmanus 
Is there an easy way to downgrade the extension version? 
I can't confirm for sure that this was happening with 0.15 but I think so. I started using vscode for C++ and working on this project in mid March (so I think the latest release by there was 0.15). The clang version was always this one since I installed clang/xcode.

EDIT:
I have installed the 0.15 and the same problem happened :(

**Comment by tekezo** [comment]
I solved this issue by this configuration.

```
{
    "C_Cpp.default.macFrameworkPath": [
        "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks",
        "/Library/Frameworks",
        "/System/Library/Frameworks"
    ]
}
```

**Comment by sherjilozair** [comment]
Problem occurs for me.
Here's the repro:
```
#include <OpenGL/gl3.h>
int main(){
  return 0;
}
```

Here's `c_cpp_properties.json`:
```
{
    "configurations": [
        {
            "name": "Mac",
            "browse": {
                "path": [
                    "${workspaceFolder}"
                ],
                "limitSymbolsToIncludedHeaders": true
            },
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [],
            "macFrameworkPath": [
                "/System/Library/Frameworks",
                "/Library/Frameworks"
            ],
            "compilerPath": "/usr/bin/clang",
            "cStandard": "c11",
            "cppStandard": "c++17",
            "intelliSenseMode": "clang-x64"
        }
    ],
    "version": 4
}
```

The OpenGL header exists in the framework path:
```
$ ls /System/Library/Frameworks/OpenGL.framework/Headers/gl3.h
/System/Library/Frameworks/OpenGL.framework/Headers/gl3.h
```

I get this error:
```
could not open source file "OpenGL/gl3.h" (no directories in search list)
```

**Comment by bobbrow** [comment]
I think I see the problem.  AE is a nested framework in the CoreServices framework.  0.17.7 should have resolved @sherjilozair's issue, but the extension is not doing recursive search of frameworks.

You can work around it by adding:
```
        "macFrameworkPath": [
            "/System/Library/Frameworks/ApplicationServices.framework/Frameworks",
            "/System/Library/Frameworks/CoreServices.framework/Frameworks"
        ]
```
to your configuration.

**Comment by sean-mcmanus** [comment]
Fixed with 0.18.0.
